### PR TITLE
docs: add more details for consumer-restriction plugin 

### PR DESCRIPTION
**Pull request by Fabriceli**
### Description

1. add more details about `allowed_by_methods` include `allowed_by_methods.user` and `allowed_by_methods.methods` in zh doc
2. add more details about `allowed_by_methods` include `allowed_by_methods.user` and `allowed_by_methods.methods` in en doc

Fixes # (issue)
#9873

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by Fabriceli** [comment]
@jiangfucheng @shreemaan-abhishek pls take a look at this PR, thanks

**Review by jiangfucheng — APPROVED** [review]
LGTM

**Comment by Fabriceli** [comment]
@monkeyDluffy6017 pls take a look at this PR, thanks

File: docs/zh/latest/plugins/consumer-restriction.md [hunk 1]
```diff
@@ -1,12 +1,6 @@
 ---
-title: consumer-restriction
-keywords:
-  - Apache APISIX
-  - API 网关
-  - Consumer restriction
-description: Consumer Restriction 插件允许用户根据 Route、Service 或 Consumer 来设置相应的访问限制。
+span
```
Comment (monkeyDluffy6017) [inline docs/zh/latest/plugins/consumer-restriction.md:2]: Why do you remove this?

Comment (monkeyDluffy6017) [reply docs/zh/latest/plugins/consumer-restriction.md:2]: @kayx23 please help to check this

Comment (Fabriceli) [reply docs/zh/latest/plugins/consumer-restriction.md]: markdownlint delete by mistake.
fixed

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**